salt/modules/debian_ip.py

```diff
@@ -1498,6 +1498,13 @@ def _write_file_routes(iface, data, folder, pattern):
     '''
     Writes a file to disk
     '''
+    # Since the entire directory is executed with matching pattern
+    # ^[a-zA-Z0-9_-]+$ not all filenames are legitimate.
+    # In order to make routes work for vlan interfaces (e.g. bond0.12)
+    # the dots in the iface name will be replaced by underscores, so it
+    # will be properly executed
```

Comment:
I'm sorry, but this doesn't seem to help with clarity. What is being executed? Why are filenames not legitimate?

Perhaps...
`The source-directory option in /etc/network/interfaces uses the pattern "^[a-zA-Z0-9_-]+$" to match filenames, which creates problems with vlan-type interfaces (e.g. bond0.12) that may have a dot in them. This replaces dots with underscores in order to maintain compatibility with both source-directory and source (with glob) file inclusions.`

Ref:
`       Similarly, "source-directory" keyword is used to source multiple  files
       at  once,  without  specifying  them individually or using shell globs.
       Additionally, when "source-directory" is used, names of the  files  are
       checked to match the following regular expression: ^[a-zA-Z0-9_-]+$. In
       other words, the names must consist entirely of ASCII upper- and lower-
       case letters, ASCII digits, ASCII underscores, and ASCII minus-hyphens.
       In the directory path, shell wildcards may be used as well.
`